### PR TITLE
Allow only users with publish permission to unpublish page

### PR DESCRIPTION
**Pull request by MizukiTemma**
### Short description

This PR tries to solve the problem that users who are not allowed to publish a page can unpublish the page by storing it as draft.

### Proposed changes

- "Save/Restore as Draft" Button will not be shown to users anymore, who do not have publishing right.
- This change is applied to Edit Page (where users edit page content) and Versions (where users manage past versions).

### Resolved issues

Fixes: #1445

### CODE REVIEW

File: integreat_cms/cms/templates/pages/page_revisions.html [hunk 1]
```diff
@@ -84,14 +84,12 @@ <h1>{{ page_translation.title }}</h1>
     <div class="w-full p-4 flex justify-end gap-4">
         {% has_perm 'cms.change_page_object' request.user page as can_edit_page %}
```
Comment (timoludwig) [inline integreat_cms/cms/templates/pages/page_revisions.html:85]:
```suggestion
```
I guess this line is not required anymore.

**Review by timoludwig — COMMENTED** [review]
Thanks a lot! :+1:
In general, the UI adjustments in the templates work as expected.
However, I think we should also check the permissions in the views to prevent users to "hack" the status by manually editing the html source code and changing e.g. `<button name="submit_review">` to `<button name="submit_draft">` and sending the faked status to the backend.
https://github.com/digitalfabrik/integreat-cms/blob/c79a54fd5abb792696420aa6427a5e5a356fa79c/integreat_cms/cms/views/pages/page_form_view.py#L256-L259

https://github.com/digitalfabrik/integreat-cms/blob/c79a54fd5abb792696420aa6427a5e5a356fa79c/integreat_cms/cms/views/pages/page_revision_view.py#L180-L181

Apart from that: Could you rebase your branch onto the develop branch as described [here](https://digitalfabrik.github.io/integreat-cms/git-flow.html#update-feature-branch) to fix the CircleCI tests? And add a changelog entry afterwards? Thanks!

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 UNRELEASED
 ----------
-
+* [ [#1445](https://github.com/digitalfabrik/integreat-cms/issues/1445) ] Allow only users with publish permission to unpublish page
```
Comment (timoludwig) [inline CHANGELOG.md:3]:
```suggestion

* [ [#1445](https://github.com/digitalfabrik/integreat-cms/issues/1445) ] Allow only users with publish permission to unpublish page

```

File: integreat_cms/cms/views/pages/page_form_view.py [hunk 3]
```diff
@@ -253,9 +253,11 @@ def post(self, request, *args, **kwargs):
...
             page_form.add_error_messages(request)
             page_translation_form.add_error_messages(request)
-        elif (
-            not request.user.has_perm("cms.publish_page_object", page_form.instance)
-            and page_translation_form.cleaned_data.get("status") == status.PUBLIC
+        elif not request.user.has_perm(
+            "cms.publish_page_object", page_form.instance
+        ) and (
+            page_translation_form.cleaned_data.get("status") == status.PUBLIC
+            or page_translation_form.cleaned_data.get("status") == status.DRAFT
```
Comment (timoludwig) [inline integreat_cms/cms/views/pages/page_form_view.py:260]:
```suggestion
            page_translation_form.cleaned_data.get("status")
            in [status.DRAFT, status.PUBLIC]
```

**Review by timoludwig — COMMENTED** [review]
Thanks!
Now, I think you need to adapt the tests to reflect the new restrictions.
E.g. here:
https://github.com/digitalfabrik/integreat-cms/blob/c79a54fd5abb792696420aa6427a5e5a356fa79c/tests/cms/views/view_config.py#L425-L446
You need to remove `AUTHOR` from the allowed roles since authors are no longer allowed to submit pages as draft.

**Comment by MizukiTemma** [comment]
Thank you for your speedy reply and help!

As I checked the issue #1520, users without publish permission can save a translation as draft in the side-by-side view too,
Should the same change be applied also for side-by-side view?

**Comment by timoludwig** [comment]
> As I checked the issue #1520, users without publish permission can save a translation as draft in the side-by-side view too,
> Should the same change be applied also for side-by-side view?

Yes, that makes sense, totally forgot about that!

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,8 @@
 UNRELEASED
 ----------
 
+* [ [#1516](https://github.com/digitalfabrik/integreat-cms/issues/1516) ] Fix save buttons alignment
```
Comment (timoludwig) [inline CHANGELOG.md:4]:
```suggestion
```
This entry is not part of this PR, but was released in version 2022.6.1.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,8 @@
...
+* [ [#1445](https://github.com/digitalfabrik/integreat-cms/issues/1445) ] Allow only users with publish permission to unpublish page
```
Comment (timoludwig) [inline CHANGELOG.md:6]:
```suggestion


```
There need to be two new lines before the next version number

**Review by timoludwig — COMMENTED** [review]
Thanks! :+1:
Looks good, just a small issue with the changelog.

**Review by timoludwig — COMMENTED** [review]
Thanks! Could  you squash your commits again? Then this PR can be merged.

**Review by timoludwig — APPROVED** [review]
:+1:
